Remove password input form for locked assessments

qreoct opened the pull request:
Solves #4532

When a student accesses a password-protected assessment before the unlock time, the password unlock field is hidden to prevent premature access of the assessment.
![image](https://user-images.githubusercontent.com/9080974/167761943-920886b3-0175-492c-a0a2-9b38148e0419.png)

ekowidianto commented on config/locales/en/course/assessment/assessments.yml:
```diff
@@ -75,7 +75,7 @@ en:
           failure: 'Could not delete assessment: %{error}'
         authenticate:
           header: 'Input Password'
-          message: 'The assessment is locked, please input the password.'
```
Could we preserve the message and add another message instead saying that the assessment is not open yet?

qreoct replied:
Done!

ekowidianto reviewed (approved):
LGTM

cysjonathan commented:
Instead of hiding the password field, could we:
1. make the input field and button disabled
2. have a tooltip to explain that these are disabled until the humanized opening datetime
3. Add a server-side check to assert assessment has opened before checking for password correctness

Later on, for future, when we move to react, we could make the client side enable the input field and button dynamically based on the current time

qreoct commented:
1. Added a more descriptive explanation for why the form is disabled
![image](https://user-images.githubusercontent.com/9080974/167984549-243fb9b0-488f-42a4-bef9-fe2604473edc.png)

2. Added a check in the `POST` request to assert assessment has opened before the password check

cysjonathan commented on config/locales/en/course/assessment/assessments.yml:
```diff
@@ -75,7 +75,8 @@ en:
         authenticate:
           header: 'Input Password'
-          message: 'The assessment is locked, please input the password.'
+          assessment_not_unlocked: 'The assessment will be unlocked on %{start_at}.'
```
This would be better written as `message_not_started` since below also uses `message_state`

qreoct replied:
Done!

cysjonathan commented on app/views/course/assessment/assessments/authenticate.html.slim:
```diff
@@ -1,10 +1,23 @@
+- locked = assessment_not_unlocked(@assessment.time_for(current_course_user))
```
This is not so much "locked" than it is "not_started"

qreoct replied:
Done!

ekowidianto commented:
Awesome and thanks for the update!

ekowidianto commented:
Awesome and thanks for the update!
